queries/src/main/java/org/apache/geode/examples/queries/Example.java

```diff
@@ -0,0 +1,173 @@
+    // connect to the locator using default port 10334
+    ClientCache cache = new ClientCacheFactory().addPoolLocator("127.0.0.1", 10334)
+        .set("log-level", "WARN").create();
+
+    // create a local region that matches the server region
+    Region<Integer, EmployeeData> region =
+        cache.<Integer, EmployeeData>createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY)
+            .create(REGIONNAME);
+
+    Example e = new Example();
+    e.accept(region);
+    cache.close();
+  }
+
+  public int getNumberOfQ1Employees() {
+    return numberOfQ1Employees;
+  }
+
+  public int getNumberOfQ2Employees() {
+    return numberOfQ2Employees;
+  }
+
+  public int getNumberOfQ3Employees() {
+    return numberOfQ3Employees;
+  }
+
+  public Map<Integer, EmployeeData> populateEmployeeData() {
+    // put entries in the hashmap
+    String[] firstNames =
+        "Alex,Bertie,Kris,Dale,Frankie,Jamie,Morgan,Pat,Ricky,Taylor,Casey,Jessie,Ryan,Skyler"
+            .split(",");
+    String[] lastNames =
+        "Able,Bell,Call,Driver,Forth,Jive,Minnow,Puts,Reliable,Tack,Catch,Jam,Redo,Skip".split(",");
+    int salaries[] = new int[] {60000, 80000, 75000, 90000, 100000};
+    int hours[] = new int[] {40, 40, 40, 40, 30, 20};
+    int emplNumber = 10000;
+
+    Map<Integer, EmployeeData> employees = new HashMap<Integer, EmployeeData>();
+    for (int index = 0; index < firstNames.length; index++) {
+      emplNumber = emplNumber + index;
+      Integer key = emplNumber;
+      String email = firstNames[index] + "." + lastNames[index] + "@example.com";
+      int salary = salaries[index % 5];
+      int hoursPerWeek = hours[index % 6];
+      EmployeeData value = new EmployeeData(firstNames[index], lastNames[index], emplNumber, email,
+          salary, hoursPerWeek);
+      employees.put(emplNumber, value);
+    }
+
+    return employees;
+  }
+
+  @Override
+  public void accept(Region<Integer, EmployeeData> region) {
+    Map<Integer, EmployeeData> employees = populateEmployeeData();
+    for (EmployeeData value : employees.values()) {
+      region.put(value.getEmplNumber(), value);
+    }
+
+    // count the values in the region
+    int inserted = region.keySetOnServer().size();
+    System.out.println(String.format("Counted %d keys in region %s", inserted, region.getName()));
+
+    // fetch and print all values in the region (without using a query)
+    region.keySetOnServer().forEach(key -> System.out.println(region.get(key)));
+    doQueries((ClientCache) region.getRegionService(), region);
+  }
+
+  // Demonstrate querying using the API.
+  private void doQueries(ClientCache cache, Region<Integer, EmployeeData> region) {
+    String regionName = region.getName();
+    QueryService queryService = cache.getQueryService();
+
+    try {
+
+      // Query for every entry in the region, and print query results.
+      // Count the number of results returned to facilitate testing.
+      numberOfQ1Employees = 0;
```

Comment:
I think this code could be made shorter and more readable by removing constants and using the for loop iterator idiom.